### DISCUSSION TITLE
Can I add columns to table with indexes ?

### DISCUSSION THREAD

**Discussion by noselasd** [duckdb/duckdb#4639] [General]
Just wondering if I'm supposed to be able to add columns to a table with indexes ?

```
alter table  "test" add column anr VARCHAR;
Error: Catalog Error: Cannot alter entry "test" because there are entries that depend on it.
```
It seems after dropping all indexes (and re-create them after), I'm able to add a column. Is that the best way to add a column, or are there better alternatives ?

**Comment by hannes** [comment]
This should eventually be supported but as you noted its not yet.
